### PR TITLE
Pawn movement

### PR DESCRIPTION
**Pull request by roht0515**
it contains the code for the movement for pawn piece and its tests

### CODE REVIEW

File: src/main/java/chess/Pawn.java [hunk 1]
```diff
@@ -3,18 +3,70 @@
...
 
+    public byte possibleMovement(final Piece targetPiece) {
+        if (targetPiece == null) {
+            return EMPTY;
+        }
+        if (this.getColor() != targetPiece.getColor()) {
+            return ENEMY;
+        }
+        return FRIEND;
+    }
```
Comment (samuelloza) [inline src/main/java/chess/Pawn.java:45]: Good job but,
I think this fragment of code is very similar to the method [isPossibleToContinue](https://github.com/JhordanSoto/ProjectChessBoysGroup/blob/3f246ea0de618c54ac17f0d466d6049d326099c9/src/main/java/chess/GetPossibleMovePiece.java#L35)
mmm this is a redundancy.

File: src/main/java/chess/Pawn.java [hunk 2]
```diff
@@ -3,18 +3,70 @@
...
+    public void changeFirstMovementState() {
+        if (firstMovement) {
+            this.firstMovement = false;
+        }
+    }
```
Comment (samuelloza) [inline src/main/java/chess/Pawn.java]: mmm I tink the "if" not is necessary

Comment (samuelloza) [reply src/main/java/chess/Pawn.java]: Reviewed your PR and I not find call for this method.

Thanks

File: src/main/java/chess/Pawn.java [hunk 3]
```diff
@@ -3,18 +3,70 @@
...
+        super.setPosition(source);
+        super.setChessBoard(chessBoard);
+        firstMovement = true;
     }
 
     public ArrayList<Position> getPossibleMoves() {
+        int row = this.getPosition().getRow();
+        int column = this.getPosition().getColumn();
         ArrayList<Position> arrayPositions = new ArrayList<Position>();
+        if ((row >= 0) && (column >= 0) && (row < LIMIT_OF_ARRAY) && (column < LIMIT_OF_ARRAY)) {
```
Comment (JhordanSoto) [inline src/main/java/chess/Pawn.java]: Try to use constants

File: src/main/java/chess/Pawn.java [hunk 4]
```diff
@@ -3,18 +3,70 @@
...
+            checkMovements(arrayPositions, "WHITE", column, row, 1, -1);
```
Comment (JhordanSoto) [inline src/main/java/chess/Pawn.java]: Try to use constants

File: src/main/java/chess/Pawn.java [hunk 5]
```diff
@@ -3,18 +3,70 @@
...
+            checkMovements(arrayPositions, "BLACK", column, row, -1, 1);
```
Comment (JhordanSoto) [inline src/main/java/chess/Pawn.java]: Try to use constants

File: src/main/java/chess/Pawn.java (hunk 1 above)
Comment (roht0515) [reply src/main/java/chess/Pawn.java:45]: sure, i don't need to add all the methods from GetPossibleMovePiece just one, this is why i created one just for the pawn class

**Pull request merged**